File: backend/app/schemas/product.py
from decimal import Decimal
from .base import BaseSchema
from .category import CategoryRef
from .station import StationRef

class ProductCreate(BaseSchema):
    name: str
    price: Decimal
    category_id: int
    station_id: int

class ProductUpdate(BaseSchema):
    name: str | None = None
    price: Decimal | None = None
    category_id: int | None = None
    station_id: int | None = None

class ProductResponse(BaseSchema):
    id: int
    name: str
    price: Decimal
    category_id: int
    station_id: int
    active: bool
    category: CategoryRef
    station: StationRef

File: frontend/src/types/product.ts
/**
 * Interface de producto
 */
import type { Category } from "./category"
import type { Station } from "./station"


export interface Product {
  id: number
  name: string
  price: number
  active: boolean
  category_id: number
  station_id: number

  category?: Pick<Category, "id" | "name">
  station?: Pick<Station, "id" | "name">
}


export interface ProductCreate {
  name: string
  price: number
  category_id: number
  station_id: number
}


export interface ProductUpdate {
  name?: string
  price?: number
  category_id?: number
  station_id?: number
}